app/src/main/java/com/duy/calculator/symja/activities/PiActivity.java

```diff
@@ -121,8 +121,43 @@ protected String getExpression() {
                     }
                 } catch (NumberFormatException nfex) {
                 }
+                newLineUtil nlu = new newLineUtil(result);
+                result = nlu.getNewLine(nlu.result);
                 return Lists.newArrayList(result);
             }
         };
     }
 }
+
+class newLineUtil {
+    String result;
+    public newLineUtil(String result){
+        this.result = result;
+    }
+
+    /**
+     * Fixs the final expression, which makes the long expression can be
+     * displayed completely, rather than some parts may hind out of the screen.
+     *
+     * @param result The ture expression result without hinden.
+     * @return ans
+     * The result after do the improvement to the result
+     * which can be changed a new line.
+     */
+    public String getNewLine(String result) {
+        int len = 21;
+        result = result.substring(2, result.length() - 2);
```

Comment:
not tested yet. But is it assured that no `IndexOutOfBoundsException` is thrown here?
- https://docs.oracle.com/javase/7/docs/api/java/lang/String.html#substring(int,%20int)

Can this method be used for other cases?

Reply:
Yes, it just use for the Latex expression to change the new line, which may cannot display on the screen, and substring method can use, and I use it to ignore the expression's "$$" at the begin and end. Thus the method not only use for the PiActivity class, it can use for all of expression result in the Latex type.
![image](https://user-images.githubusercontent.com/72643295/119233683-c74c8c00-bb5c-11eb-8dc8-c58311b4ce54.png)